shared/extd_nw_types.py

```diff
@@ -3,21 +3,257 @@
 Defines and Implements Basic Network packet types, such as TCP and UDP.
 """
 
+from pyof.foundation.network_types import IPv4, IPv6
 from pyof.foundation.base import GenericStruct
-# from pyof.foundation.basic_types import (BinaryData, UBInt16)
-
+from pyof.foundation.basic_types import BinaryData, UBInt16, UBInt32
+from typing import Union
+from random import randrange
 
 __all__ = ('TCP', 'UDP')
 
 
 class TCP(GenericStruct):
-    """TCP - To be developed """
-    def __init__(self):
-        pass
+    """TCP - Being developed """
+
+    source_p = UBInt16()
+    destination_p = UBInt16()
+    sequence_n = UBInt32()
+    acknowledgment_n = UBInt32()
+    _length_flags = UBInt16()
+    window = UBInt16()
+    checksum = UBInt16()
+    urgent_pointer = UBInt16()
+    options = BinaryData()
+    data = BinaryData()
+
+    def __init__(self, source_p:int=0, destination_p:int=0, sequence_n:int=0, acknowledgment_n:int=0, h_length:int=5,
```

Comment:
When you assign the default integer value no need to also type hint with `:int` that 's derived implicitly. Although the intention was good, feel free to remove it.

Reply:
Removed.